bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/InitializeToolsStartup.java

```diff
@@ -165,6 +196,44 @@ public void run()
 		}
 	}
 
+	private void checkForUpdatedVersion(File idf_json_file)
+	{
+		// read esp-idf.json file
+		JSONParser parser = new JSONParser();
+		try
+		{
+			JSONObject jsonObj = (JSONObject) parser.parse(new FileReader(idf_json_file));
```

Comment:
It's better and safer to use try-with-resources with a reader , which implements AutoClosable. The try-with-resources statement ensures that each resource is closed at the end of the statement.